src/v2/view-builder/components/BaseAuthenticatorView.js

```diff
@@ -1,18 +1,23 @@
+import { _ } from 'okta';
 import { BaseHeader, BaseView } from '../internals';
 import AuthenticatorFooter from './AuthenticatorFooter';
 import HeaderBeacon from './HeaderBeacon';
 import { getIconClassNameForBeacon } from '../utils/AuthenticatorUtil';
 
-const HeaderBeaconFactor = HeaderBeacon.extend({
+export const BaseAuthenticatorBeacon = HeaderBeacon.extend({
+  authenticatorKey() {
+    return this.options.appState.get('authenticatorKey');
+  },
+
   getBeaconClassName: function() {
-    const authenticatorKey = this.options.appState.get('authenticatorKey');
+    const authenticatorKey = _.result(this, 'authenticatorKey');
```

Comment:
`nit`: This is equivalent to saying `this.get('authenticatorKey') || this.authenticatorKey()` right? But looks like the code is passing `authenticatorKey` in options? So shouldn't it be referring to `this.options.authenticatorKey` instead of `this.get('authenticatorKey')`? If they both are same, this is good to go..

Reply:
This is equivalent
```
IF this.authenticatorKey is function then return this.authenticatorKey();
ELSE return this.authenticatorKey
```